src/web/mjs/connectors/MangaParkEN.mjs

```diff
@@ -79,22 +171,112 @@ export default class MangaParkEN extends Connector {
+                case 'zh_hk':
+                    language = 'zh-Hans';
+                    break;
+                case 'zh_tw':
+                    language = 'zh-Hant';
+                    break;
+                case 'pt_br':
+                    language = 'pt-BR';
+                    break;
+                case 'es_419':
+                    language = 'es-419';
+                    break;
+                case '_t':
+                    language = 'other';
+                    break;
+            }
+            return {
+                id: source.data.id,
+                lang: language,
+                srcTitle: source.data.srcTitle
+            };
+        });
+    }
+
+    async _getChaptersFromSource(source) {
+        const gql = `
+            query get_content_source_chapterList($sourceId: Int!) {
+                get_content_source_chapterList(sourceId: $sourceId) {
+                    data {
+                        id, dateCreate, dname, title, urlPath
                     }
-                } catch(error) {
-                    reject(error);
                 }
-            }, 2500);
+            }
+        `;
+        const vars = {"sourceId": source.id};
+        const data = await this.fetchGraphQL(this.apiURL, 'get_content_source_chapterList', gql, vars);
+        return data.get_content_source_chapterList.map(chapter => {
+            return {
+                id: chapter.data.urlPath,
+                title: chapter.data.dname + (chapter.data.title == null || chapter.data.title.length == 0 ? '' : ` - ${chapter.data.title}`) + ` (${source.lang}) [${source.srcTitle}]`,
+                language: source.lang,
+                creationDate: chapter.data.dateCreate
+            };
         });
+    }
+
+    async _getPages(chapter) {
+        const script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    try {
+                        if(typeof app !== 'undefined' && typeof app.items !== 'undefined') {
+                            resolve(app.items.map(item => item.src || item.isrc));
+                        } else if (typeof amWord !== 'undefined' && typeof amPass !== 'undefined' && typeof imgHostLis !== 'undefined' && typeof imgPathLis !== 'undefined') {
+                            const params = JSON.parse(CryptoJS.AES.decrypt(amWord, amPass).toString(CryptoJS.enc.Utf8));
+                            resolve(imgHostLis.map((data, i) => \`\${data}\${imgPathLis[i]}?\${params[i]}\`));
+                        } else {
+                            const images = __NEXT_DATA__.props.pageProps.dehydratedState.queries
+                                .map(query => {
+                                    const { httpLis, wordLis } = query.state.data.data.imageSet;
+                                    return httpLis.map((path, i) => \`\${path}?\${wordLis[i]}\`);
+                                });
+                            resolve([].concat(...images));
```

Comment:
Consider just returning the `images`, no need to spread and concat to empty array.

Reply:
If I simply return images, I return an array of array of strings - `[ [ 'link1', 'link2' ] ]`.

I can take first parsed query after `map` (i.e. array).
But what if there are several queries for pages in some chapter in the future?

Reply:
I wasn't aware `images` is a nested array